astra/src/main/java/com/slack/astra/chunk/ReadOnlyChunkImpl.java

```diff
@@ -433,11 +435,11 @@ private void handleChunkAssignment(CacheSlotMetadata cacheSlotMetadata) {
           TimeUnit.SECONDS.convert(durationNanos, TimeUnit.NANOSECONDS),
           FileUtils.byteCountToDisplaySize(FileUtils.sizeOfDirectory(dataDirectory.toFile())));
     } catch (Exception e) {
-      // if any error occurs during the chunk assignment, try to release the slot for re-assignment,
-      // disregarding any errors
-      setChunkMetadataState(cacheSlotMetadata, Metadata.CacheSlotMetadata.CacheSlotState.FREE);
-      LOG.error("Error handling chunk assignment", e);
+      LOG.error("Error handling chunk assignment: {}\n{}", e, cacheSlotMetadata);
```

Comment:
Maybe try
```suggestion
      LOG.error("Error handling chunk assignment for cache slot {}:", cacheSlotMetadata, e);
```
so that the exception is logged with all its metadata.

https://www.slf4j.org/faq.html#paramException